### PR TITLE
fix curve api

### LINKED ISSUE #205: Improve Curve API
- https://github.com/0xfe/vexflow/issues/1607 (I think this is a big ask)

_Originally posted by @jaredjj3 in https://github.com/vexflow/vexflow/issues/176#issuecomment-1978655914_

### PR DESCRIPTION
**Pull request by rvilarl**
fixes #205 bullet 1 & 2 I do not know how to proceed with 3.

New test for opening direction

image

### CODE REVIEW

**Comment by rvilarl** [comment]
@jaredjj3 could you also have a look?

File: tests/curve_tests.ts [hunk 1]
```diff
@@ -83,6 +84,36 @@ const simple = createTest(
...
+  (f, notes) => {
+    f.Curve({
+      from: notes[0],
+      to: notes[3],
+      options: {
+        cps: [
+          { x: 0, y: 10 },
+          { x: 0, y: 50 },
+        ],
+        openingDirection: 'up',
```
Comment (jaredjj3) [inline tests/curve_tests.ts:99]: In the test image, it looks like it curves up then down. Does it look this way without those `cps`?

It seems like in order to honor `openingDirection`, vexflow will have to control `cps`. What have you found when experimenting with it?

Comment (rvilarl) [reply tests/curve_tests.ts:99]: Without the cps on the first curve

image

Comment (rvilarl) [reply tests/curve_tests.ts:99]: I have not experimented much :)

**Comment by rvilarl** [comment]
@jaredjj3 is this PR fine?

**Comment by jaredjj3** [comment]
While the API is what I asked for, it doesn't seem to guarantee the desired effect because of the issue with `cps`. Is there a way you can guarantee that the combination of `(openingDirection, invert)` options are honored?

**Comment by rvilarl** [comment]
> While the API is what I asked for, it doesn't seem to guarantee the desired effect because of the issue with `cps`. 

@jaredjj3 could you provide a drawing of the wished result with up and down?
> Is there a way you can guarantee that the combination of `(openingDirection, invert)` options are honored?

yes, up + invert results in down and down + invert results in up

**Comment by jaredjj3** [comment]
This is what I mean. However, I think `openingDirection` would be obsolete if `vexflow` suggested `cps` that rendered in a way that avoids bounding boxes of other tickables. I speculate that people would not want the second measure as an option.

image

Do you think `cps` just has to be manually adjusted by a human? After all, the [<slur> MusicXML](https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/slur/) encodes that information.

**Comment by rvilarl** [comment]
> This is what I mean. However, I think `openingDirection` would be obsolete if `vexflow` suggested `cps` that rendered in a way that avoids bounding boxes of other tickables. I speculate that people would not want the second measure as an option.
> 
> image

I assumed 'up' and 'down' in the other direction, now fixed. I have produced to tests one for 'up'

image

and one for 'down'

image

> Do you think `cps` just has to be manually adjusted by a human? After all, the [ MusicXML](https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/slur/) encodes that information.

I think that the fact that  musicxml is storing it, indicates manual adjustment done in the edition tool.

@jaredjj3 is that now looking good?

**Comment by jaredjj3** [comment]
@rvilarl, yes! This is exactly what I meant. This should let me control slur directions in vexml. LGTM

**Pull request merged**